src/Java.Interop/Java.Interop/JniRuntime.JniValueManager.cs

```diff
@@ -271,6 +271,28 @@ static Type GetPeerType ([DynamicallyAccessedMembers (Constructors)] Type type)
 				return type;
 			}
 
+			public IJavaPeerable? GetPeer (
+					JniObjectReference reference,
+					[DynamicallyAccessedMembers (Constructors)]
+					Type? targetType = null)
+			{
+				if (disposed) {
+					throw new ObjectDisposedException (GetType ().Name);
+				}
+
+				if (!reference.IsValid) {
+					return null;
+				}
+
+				var peeked  = PeekPeer (reference);
+				if (peeked != null &&
+						(targetType == null ||
+						 targetType.IsAssignableFrom (peeked.GetType ()))) {
+					return peeked;
+				}
+				return CreatePeer (ref reference, JniObjectReferenceOptions.Copy, targetType);
+			}
```

Comment:
I was able to update `Java.Lang.Object.GetObject()` to use this method:
```csharp
var p = JNIEnvInit.ValueManager!.GetPeer (new JniObjectReference (handle), type);
JNIEnv.DeleteRef (handle, transfer);
return p;
```
After merging this, we can try it in a new version of:
* https://github.com/dotnet/android/pull/9630